core/feconf.py

```diff
@@ -24,11 +24,31 @@
 
 from core.constants import constants
 
-from typing import Dict, List, Union
+from typing import Dict, List
 from typing_extensions import TypedDict
 
-CommandType = (
-    Dict[str, Union[str, List[str], Dict[str, Union[str, List[str]]]]])
+
+# This TypedDict is used only to create the TypedDict below, because we want
+# to mark some attributes as optional.
+class _ValidCmdDictNotTotal(TypedDict, total=False):
+    """Dictionary representing a non-required part
+    of valid commands specs."""
+
+    allowed_values: Dict[str, List[str]]
+    deprecated_values: Dict[str, List[str]]
+
+
+class ValidCmdDict(
+    _ValidCmdDictNotTotal,
+    total=False
+):
```

Comment:
This can now maybe fit one line? or three lines?

Reply:
Done!